subsys/bluetooth/audio/Kconfig.ascs

```diff
@@ -35,4 +35,13 @@ config BT_ASCS_ASE_SNK
 config BT_ASCS_ASE_SRC
 	def_bool BT_ASCS_ASE_SRC_COUNT > 0
 	select BT_PAC_SRC
+
+config BT_ASCS_MAX_ACTIVE_ASES
+    int "Number of simultaneously supported ASE sessions" 
+    default BT_MAX_CONN
```

Comment:
I think that depends: We can in theory have more than `BT_ISO_MAX_CHAN` ASEs, as long as they are not in the streaming state. The number of ASEs for a single connection we can support will be equal to the number of `bt_audio_iso` we support, but it is also a bit tricky as we have an ASE per connection (unlike the number of `bt_audio_iso` or `BT_ISO_MAX_CHAN` which are a total, and also includes the number of broadcast ISO streams supported). 

I don't see any reason why we would restrict the number of ASEs to any value (the number of active ASEs will depend on the use case). 

Since this will add an ASE pool, I think defaulting to 1 ASE per connection we support works OK.

Reply:
The max is clearly wrong below though, it should be
(BT_ASCS_ASE_SNK_COUNT + BT_ASCS_ASE_SRC_COUNT) * BT_MAX_CONN
Which also would be what I'd prefer to put the default value to above.
Therefor I ask you @Thalley @MariuszSkamra  what you prefer the default to be

Reply:
I'm not sure how to best solve this, since as I understand we cannot use arithmetics in kconfig:

```
config BT_ASCS_MAX_ACTIVE_ASES
	int "Number of simultaneously supported ASE sessions"
	default 2       # BT_MAX_CONN defaults to 1, so we default to two active ases per connection
	range 1 127500  # Highest range of: (BT_MAX_CONN * (BT_ASCS_ASE_SNK_COUNT + BT_ASCS_ASE_SRC_COUNT))
	help
	  The number of simultanesouly supported ASE sessions, in particular
	  meaning the number of ASE State Machines that are allowed to be in a
	  non-IDLE state at a single time.
```

Reply:
We should also consider if we are overengineering this :D While we technically can have more ASEs "active" than we support ISO channels, once an ASE enters the QoS configured state it needs an audio_iso (which is limited to the number of ISO chans we support), so we could probably also just put the number of active ASEs to be the same as number of active ISO chans. 

The only use case for keeping an ASE only in the codec configured state, is to support the caching, but then we need to use the max of number of conn and bonds we support.

Reply:
Yes, this is a really good point.

Also, we don't really support caching today if I interpret it correct. Unless the peer has disconnected us, we will always disconnects and go to IDLE.

Reply:
With that in mind I set the default, and max to BT_ISO_MAX_CHAN. As long as we don't support caching that makes sense to me.

Reply:
Just keep in mind that `BT_ISO_MAX_CHAN` may be larger than `BT_MAX_CONN * (BT_ASCS_ASE_SNK_COUNT + BT_ASCS_ASE_SRC_COUNT)` :D

Reply:
I added a BUILD_ASSERT for this expressions instead, and changed the range to an integer literal.